prow/sidecar/run.go

```diff
@@ -121,13 +122,40 @@ func (o Options) Run() error {
 		aborted = returnCode == 130
 	}
 
-	return o.doUpload(spec, passed, aborted)
+	metadataFile := o.WrapperOptions.MetadataFile
+	if _, err := os.Stat(metadataFile); err != nil {
+		if !os.IsNotExist(err) {
+			logrus.WithError(err).Errorf("Failed to stat %s", metadataFile)
+		}
+		return o.doUpload(spec, passed, aborted, nil)
+	}
+
+	metadataRaw, err := ioutil.ReadFile(metadataFile)
+	if err != nil {
+		logrus.WithError(err).Errorf("cannot read %s", metadataFile)
+		return o.doUpload(spec, passed, aborted, nil)
+	}
+
+	metadata := map[string]string{}
```

Comment:
Is there any need to enforce a string->string map instead of allowing an arbitrary JSON object? It would be nice to be able to include arrays, numbers, and objects in the metadata.

Reply:
will switch to an interface